src/main/java/uk/gov/companieshouse/api/interceptor/MappedCRUDAuthenticationInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.api.interceptor;
+
+import org.springframework.http.HttpMethod;
+import uk.gov.companieshouse.api.util.security.Permission;
+import uk.gov.companieshouse.api.util.security.TokenPermissions;
+
+/**
+ * Replacement for {@link CRUDAuthenticationInterceptor} implemented with
+ * {@link MappablePermissionsInterceptor}.
+ * Checks the request contains the relevant token permission values based on the HTTP method.
+ * It will try to find a {@link TokenPermissions} object in the request or create one and store
+ * it in the request if not.
+ */
+public class MappedCRUDAuthenticationInterceptor extends MappablePermissionsInterceptor {
+    private static final PermissionsMapping CRUD_MAPPING = PermissionsMapping.builder()
+            .defaultAllOf(Permission.Value.READ)
+            .mapAllOf(HttpMethod.PUT.toString(), Permission.Value.UPDATE)
+            .mapAllOf(HttpMethod.PATCH.toString(), Permission.Value.UPDATE)
+            .mapAllOf(HttpMethod.POST.toString(), Permission.Value.CREATE)
```

Comment:
Are these mapping as default permissions for these methods? 
Unless I've mistaken the purpose of this code -
I see why it could be seen as prudent to add them as it seems fairly natural - surely a POST is a create, right?
However, in this instance, we could be adding unnecessary allowed permissions - ROA for instance will only ever check for 'update' permission and will do a POST to create the submission resource. So whilst a user shouldn't ever actually be able to be prescribed a 'create' permission because it's not in the ScopeMapper in the Account service, that's still not an accurate reflection of the permissions that the user needs to perform the tasks against that service.
Since this is a mappable interceptor, let the services who know what they are checking against to map all of it.

Reply:
To clarify my intention of the code above, it's to reproduce the exact same logic as `CRUDAuthenticationInterceptor.getValue()`:

- PUT or PATCH require user's permissions to contain 'update'
- POST requires user's permissions to contain 'create'
- any other HTTP method requires user's permissions to contain 'read'

(to be added in the fix)
- DELETE requires user's permissions to contain 'delete'